include/envoy/http/filter.h

```diff
@@ -847,18 +857,29 @@ class FilterChainFactoryCallbacks {
    */
   virtual void addStreamDecoderFilter(Http::StreamDecoderFilterSharedPtr filter) PURE;
 
+  virtual void addStreamDecoderFilter(Http::StreamDecoderFilterSharedPtr filter,
+                                      MatchTreeSharedPtr match_tree,
+                                      MatchingDataPtr matching_data) PURE;
```

Comment:
Can you add comments? I'm not sure why we are passing new match_tree, matching_data, etc.